merlin/models/torch/models/ranking.py

```diff
@@ -74,3 +78,76 @@ def __init__(
         )
 
         super().__init__(dlrm_body, output_block)
+
+
+class DCNModel(Model):
+    """
+    The Deep & Cross Network (DCN) architecture as proposed in Wang, et al. [1]
+
+    Parameters
+    ----------
+    schema : Schema
+        The schema to use for selection.
+    depth : int, optional
+        Number of cross-layers to be stacked, by default 1
+    deep_block : Block, optional
+        The `Block` to use as the deep part of the model (typically a `MLPBlock`)
+    stacked : bool
+        Whether to use the stacked version of the model or the parallel version.
+    input_block : Block, optional
+        The `Block` to use as the input layer. If None, a default `TabularInputBlock` object
+        is instantiated, that creates the embedding tables for the categorical features
+        based on the schema. The embedding dimensions are inferred from the features
+        cardinality. For a custom representation of input data you can instantiate
+        and provide a `TabularInputBlock` instance.
+
+    Returns
+    -------
+    Model
+        An instance of Model class representing the fully formed DCN.
+
+    Example usage
+    -------------
+    >>> model = mm.DCNModel(
+    ...    schema,
+    ...    depth=2,
+    ...    deep_block=mm.MLPBlock([256, 64]),
+    ...    output_block=BinaryOutput(ColumnSchema("target")))
```

Comment:
Should we try to make our examples black-compatible? In that case I guess it would put the last `)` on a new line